src/Build.OM.UnitTests/Construction/SolutionFile_Tests.cs

```diff
@@ -1053,22 +1215,37 @@ public void ParseSolutionWithComments()
 
         /// <summary>
         /// Helper method to create a SolutionFile object, and call it to parse the SLN file
-        /// represented by the string contents passed in.
+        /// represented by the string contents passed in. Optionally can convert the SLN to SLNX and then parse the solution.
         /// </summary>
-        private static SolutionFile ParseSolutionHelper(string solutionFileContents)
+        private static SolutionFile ParseSolutionHelper(string solutionFileContents, bool convertToSlnx = false)
         {
             solutionFileContents = solutionFileContents.Replace('\'', '"');
             string solutionPath = FileUtilities.GetTemporaryFileName(".sln");
-
+            string slnxPath = solutionPath + "x";
             try
             {
                 File.WriteAllText(solutionPath, solutionFileContents);
-                SolutionFile sp = SolutionFile.Parse(solutionPath);
-                return sp;
+                if (convertToSlnx)
+                {
+                    ISolutionSerializer serializer = SolutionSerializers.GetSerializerByMoniker(solutionPath);
+                    SolutionModel solutionModel = serializer.OpenAsync(solutionPath, CancellationToken.None).Result;
+                    SolutionSerializers.SlnXml.SaveAsync(slnxPath, solutionModel, CancellationToken.None).Wait();
+
+                    SolutionFile slnx = SolutionFile.Parse(slnxPath);
+                    return slnx;
+                }
+
+                SolutionFile sln = SolutionFile.Parse(solutionPath);
+                return sln;
             }
             finally
             {
                 File.Delete(solutionPath);
+
+                if (convertToSlnx)
+                {
+                    File.Delete(slnxPath);
```

Comment:
Ideally we'd handle this kind of cleanup with our `TestEnvironment`, handling this with something like an `ExpectFile` that will automatically clean up on dispose. But no need to refactor now.